### PR TITLE
Show html media in gallery list as static image

### LINKED ISSUE #7535: Show html media in gallery list as static image
### Is your feature request related to a problem?

Performance and UI wise it is not smart to load possibly big html/css/js iframe on the list, which should be fast.

### Describe the solution you would like

When media is type of `text/hmtl` i would show just `image`

### Describe alternatives you have considered

Possibly we can add some small icon? or special effect around the card which would mark it as interactive? cc @exe

### Screenshots

Screenshot 2023-10-06 at 12 09 39

### Are there opened related issues?

_No response_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Feature
  - [x] Refactoring

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #7535
  - [x] refactor: unify media resolver

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

image

  ## Copilot Summary
  
### <samp>🤖 Generated by Copilot at a667ddc</samp>

This pull request replaces the `MediaResolver` component and its subcomponents with the `MediaItem` component from the `@kodadot1/brick` library. It also adds new SCSS classes for styling UI elements with theme variables. These changes improve the consistency, performance, and interactivity of media rendering and previewing in the gallery.

  
### <samp>🤖 Generated by Copilot at a667ddc</samp>

> _We're sailing on the web with `MediaItem` in our hand_
> _We've styled our borders, backgrounds, shapes, and text with SCSS_
> _We've refactored our components to make them smart and fast_
> _So heave away, me hearties, heave away on the count of three_

### CODE REVIEW

File: assets/styles/global.scss [hunk 1]
```diff
@@ -501,3 +514,11 @@ a.has-text-grey {
...
   border-left: none !important;
 }
+
+.is-rounded {
+  border-radius: 50%;
+}
+
+.is-text-weight-500 {
+  font-weight: 500;
+}
```
Comment (roiLeo) [inline assets/styles/global.scss:524]: `has-text-weight-medium`

File: libs/ui/src/components/MediaItem/MediaItem.vue [hunk 2]
```diff
@@ -23,6 +23,13 @@
         $t('lewd.explicitDesc')
       }}</span>
     </div>
+    <div
+      v-if="isInteractive"
+      class="k-shade border-k-grey is-flex is-align-items-center is-justify-content-center border is-rounded absolute-position image is-24x24">
+      <div class="is-size-7 is-text-weight-500">
+        {{ '<' + '/' + '>' }}
```
Comment (vikiival) [inline libs/ui/src/components/MediaItem/MediaItem.vue:30]: Move to some `const`

File: libs/ui/src/components/MediaItem/MediaItem.vue [hunk 3]
```diff
@@ -90,13 +101,19 @@ const components = {
...
 const resolveComponent = computed(() => {
-  return components[resolveMedia(mimeType.value) + SUFFIX]
+  let mediaType = resolveMedia(mimeType.value)
+
+  if (mediaType === MediaType.IFRAME && !props.isDetail) {
+    isInteractive.value = true
+    mediaType = MediaType.IMAGE
+  }
+  return components[mediaType + SUFFIX]
 })
```
Comment (vikiival) [inline libs/ui/src/components/MediaItem/MediaItem.vue:111]: 821wmj

**Review by vikiival — COMMENTED** [review]
@kodadot/qa-guild 🥺

File: libs/ui/src/components/MediaItem/MediaItem.vue (hunk 3 above)
Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/MediaItem.vue:111]: Except for the NFT details page, all other iframe would be displayed as images

File: assets/styles/global.scss (hunk 1 above)
Comment (Jarsen136) [reply assets/styles/global.scss]: DONE

File: libs/ui/src/components/MediaItem/MediaItem.vue (hunk 2 above)
Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/MediaItem.vue]: DONE

File: libs/ui/src/components/MediaItem/MediaItem.vue [hunk 4]
```diff
@@ -26,8 +26,8 @@
     <div
       v-if="isInteractive"
       class="k-shade border-k-grey is-flex is-align-items-center is-justify-content-center border is-rounded absolute-position image is-24x24">
-      <div class="is-size-7 is-text-weight-500">
-        {{ '<' + '/' + '>' }}
+      <div class="is-size-7 has-text-weight-medium">
+        {{ interactiveIconTitle }}
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/MediaItem.vue:30]: or use icon [code](https://fontawesome.com/icons/code?f=classic&s=solid)?

Comment (Jarsen136) [reply libs/ui/src/components/MediaItem/MediaItem.vue]: Thank you. It also works.

**Review by roiLeo — APPROVED** [review]

File: libs/ui/src/components/MediaItem/MediaItem.vue (hunk 3 above)
Comment (vikiival) [reply libs/ui/src/components/MediaItem/MediaItem.vue:112]: Amazing!

**Comment by yangwao** [comment]
pay 50 usd 
let's go

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 3.83 USD/DOT ~ 13.055 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x27ab2b75278f6f8d5c6708a18733a30ad90718b2bd3c72f39fa9e12ef763ee0c](https://polkadot.subscan.io/extrinsic/0x27ab2b75278f6f8d5c6708a18733a30ad90718b2bd3c72f39fa9e12ef763ee0c)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Review by yangwao — APPROVED** [review]

**Pull request merged**